sql: minor EXPLAIN ANALYZE cleanups

RaduBerinde opened the pull request:
#### sql: move top-level explain fields to explain package

The explain package contains all field names, except the top-level
fields (like "distribution", "planning time"). Move them to this
package by providing specific methods to set them.

Release note: None

#### sql: add MakeDeterministic explain flag

This change adds a flag that causes editing of non-deterministic
values like planning time and even distribution/vectorization. This
allows testing of EXPLAIN ANALYZE (PLAN) output with all logic test
configs.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/56844)
<!-- Reviewable:end -->

asubiotto reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 3 of 3 files at r1, 6 of 6 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56844#-:-MMR1p6DCG14ajepSM8h:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

RaduBerinde committed:
sql: move top-level explain fields to explain package

The explain package contains all field names, except the top-level
fields (like "distribution", "planning time"). Move them to this
package by providing specific methods to set them.

Release note: None

RaduBerinde committed:
sql: add MakeDeterministic explain flag

This change adds a flag that causes editing of non-deterministic
values like planning time and even distribution/vectorization. This
allows testing of EXPLAIN ANALYZE (PLAN) output with all logic test
configs.

Release note: None

mgartner reviewed (approved):
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

> Reviewed 1 of 3 files at r1, 8 of 9 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56844#-:-MMRm0HK6-jR4tOdA_W6:b-uwkowd)* status: :shipit: complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @asubiotto and @RaduBerinde)

___
*[pkg/sql/opt/exec/explain/output.go, line 307 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/56844#-MMRlnQOBNnwuEvbEbAb:-MMRlnQOBNnwuEvbEbAc:b-goacnc) ([raw file](https://github.com/cockroachdb/cockroach/blob/a88e8e78dfa45beea20753689d448d04ba78bc54/pkg/sql/opt/exec/explain/output.go#L307)):*
> ```Go
> // AddTopLevelField adds a top-level field. Cannot be called while inside a
> // node.
> func (ob *OutputBuilder) AddTopLevelField(key, value string) {
> ```

[nit] This could be `addTopLevelField`, unless you'll use this outside the package in the future.


<!-- Sent from Reviewable.io -->

RaduBerinde reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56844)* status: :shipit: complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @asubiotto and @mgartner)

___
*[pkg/sql/opt/exec/explain/output.go, line 307 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/56844#-MMRlnQOBNnwuEvbEbAb:-MMRqOFx7eiMUnEJ95SG:b-kg8d8b) ([raw file](https://github.com/cockroachdb/cockroach/blob/a88e8e78dfa45beea20753689d448d04ba78bc54/pkg/sql/opt/exec/explain/output.go#L307)):*
<details><summary><i>Previously, mgartner (Marcus Gartner) wrote…</i></summary><blockquote>

[nit] This could be `addTopLevelField`, unless you'll use this outside the package in the future.
</blockquote></details>

For now, it's consistent with a lot of the other stuff like AddField, Attr, etc. I think once the design solidifies it will probably make sense to unexport everything that has a `key` argument.


<!-- Sent from Reviewable.io -->

RaduBerinde reviewed (commented):
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56844#-:-MMRqc2a5ApRvTUGektm:b-ydi9ay)* status: :shipit: complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @asubiotto and @mgartner)



<!-- Sent from Reviewable.io -->

mgartner commented:
___
*[pkg/sql/opt/exec/explain/output.go, line 307 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/56844#-MMRlnQOBNnwuEvbEbAb:-MMbrHa48xLLSdvMWSvx:b-k2g08m) ([raw file](https://github.com/cockroachdb/cockroach/blob/a88e8e78dfa45beea20753689d448d04ba78bc54/pkg/sql/opt/exec/explain/output.go#L307)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

For now, it's consistent with a lot of the other stuff like AddField, Attr, etc. I think once the design solidifies it will probably make sense to unexport everything that has a `key` argument.
</blockquote></details>

SGTM :thumbsup: 


<!-- Sent from Reviewable.io -->
